backends/p4tools/modules/testgen/core/compiler_target.h

```diff
@@ -0,0 +1,41 @@
+#ifndef BACKENDS_P4TOOLS_MODULES_TESTGEN_CORE_COMPILER_TARGET_H_
+#define BACKENDS_P4TOOLS_MODULES_TESTGEN_CORE_COMPILER_TARGET_H_
+
+#include "backends/p4tools/common/compiler/compiler_target.h"
+#include "backends/p4tools/common/compiler/reachability.h"
+#include "midend/coverage.h"
+
+namespace P4Tools::P4Testgen {
+
+/// Extends the CompilerResult with the associated P4RuntimeApi
+class TestgenCompilerResult : public CompilerResult {
+ private:
+    /// The coverabled Nodes in the analyzed P4 program.
+    P4::Coverage::CoverageSet coverableNodes;
+
+    /// The call graph of the analyzed P4 program.
+    const NodesCallGraph *callGraph;
+
+ public:
+    explicit TestgenCompilerResult(CompilerResult compilerResult,
+                                   P4::Coverage::CoverageSet coverableNodes,
+                                   const NodesCallGraph *callGraph = nullptr);
+
+    /// @returns the call graph of the analyzed P4 program.
```

Comment:
Since this is valid only in some cases, it would make sense to mention that in the comment.